catalog-rest-service/src/main/java/org/openmetadata/catalog/events/ChangeEventHandler.java

```diff
@@ -137,22 +137,24 @@ private void handleWebSocket(ContainerResponseContext responseContext) {
         String jsonThread = mapper.writeValueAsString(thread);
         switch (thread.getType()) {
           case Task:
-            List<EntityReference> assignees = thread.getTask().getAssignees();
-            assignees.forEach(
-                (e) -> {
-                  if (Entity.USER.equals(e.getType())) {
-                    WebSocketManager.getInstance()
-                        .sendToOne(e.getId(), WebSocketManager.taskBroadcastChannel, jsonThread);
-                  } else if (Entity.TEAM.equals(e.getType())) {
-                    // fetch all that are there in the team
-                    List<EntityRelationshipRecord> records =
-                        dao.relationshipDAO()
-                            .findTo(e.getId().toString(), TEAM, Relationship.HAS.ordinal(), Entity.USER);
-                    WebSocketManager.getInstance()
-                        .sendToManyWithString(records, WebSocketManager.taskBroadcastChannel, jsonThread);
-                  }
-                });
-            return;
+            if (thread.getPostsCount() == 0) {
```

Comment:
@mohitdeuex why are we checking thread.getPostsCount() to be 0

Reply:
@harshach  Problem is that task related conversation are also posted with Thread type as "Task", it make the task related conversation fall into TASK notification channel , When we create a new Task , there are no posts attached to that thread , hence this check assure that we have a new thread , then only we post it on taskChannel, else we have post attached to that thread , the flow needs to fall into Conversation .

Reply:
@vivekratnavel can't we add a type to thread to make it a task?

Reply:
@harshach Thread has a type, and this switch case handles "Task" type of threads. I think @mohitdeuex checks for postsCount to make sure we send the notification only when the task gets created and not every time when someone replies to the task thread.

Reply:
we should do it every time someone replies right. It is indeed a subscribed Task if its assigned to me or I am the assignee

Reply:
I agree. @mohitdeuex Whenever there is an update to the task that a user is assigned, it makes sense to send them a notification with the TASK notification channel. What is the motivation for this PR?